### PR TITLE
Docs refactoring

### PR DESCRIPTION
**Pull request by navalev**
This PR proposes refactoring of the main readme and some of the deployment readme files

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -9,286 +9,91 @@
...
-
-3. Streaming platforms
-    * Kafka
-    * Azure Events Hubs
-
-4. REST requests
-
-It then can export the results to file storage, databases or streaming platforms.
+- [Setting up a development environment](docs/development.md)
+- [Run locally with KIND (Kubernetes in Docker)](docs/KIND.md)
```
Comment (tamirkamara) [inline README.MD:52]: This might need to be in the deploy doc. what do you think?

Comment (navalev) [reply README.MD:52]: I wanted to seperate between development or test deployment and production deployment. So kept this in a seperate document. Do you suggest combining all into one deployment doc with sub sections for test and production?

File: README.MD [hunk 2]
```diff
@@ -9,286 +9,91 @@
...
 :warning: Presidio can help identify sensitive/PII data in un/structured text. However, because Presidio is using trained ML models, there is no guarantee that Presidio will find all sensitive information. Consequently, additional systems and protections should be employed.
 
-## Demo
-
-[Try Presidio with your own data](https://aka.ms/presidio-demo)
-
-## Features
-
-***Unstructured text anonymization***
+[Presidio Features](docs/overview.md#features)
```
Comment (omri374) [inline README.MD:21]: The overview file is quite old, can we use the main README instead? I think they have a lot of overlapping content

Comment (omri374) [reply README.MD:21]: IMHO the high-level features should be on the main Readme

File: README.MD [hunk 3]
```diff
@@ -9,286 +9,91 @@
...
-[![Image1](docs/assets/before-after.png)](docs/assets/before-after.png)
-
-For each PII entity, presidio returns a confidence score:
-
-[![Image2](docs/assets/findings.png)](docs/assets/findings.png)
-
-***Text anonymization in images*** (beta)
-
-Presidio uses OCR to detect text in images. It further allows the redaction of the text from the original image.
+## Demo
```
Comment (omri374) [inline README.MD:25]: Why not show the different images? To me graphics really help understand what a tool does.

File: README.MD (hunk 1 above)
Comment (omri374) [reply README.MD:52]: I think we should go with one of two options:
1. Have all different deployment options (either test or prod) in one file, and reference those that are relevant for development from the develop doc. 
2. Have only the K8S deployment in 'deploy', and the rest under 'develop'

WDYT?

File: docs/KIND.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Run Presidio locally with KIND
```
Comment (omri374) [inline docs/KIND.md:1]: Where is this file referenced?

File: docs/samples.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
+# Presidio API Samples
```
Comment (omri374) [inline docs/samples.md:1]: There's an index.md file with links to all documents. This allows us to expose this doc: https://microsoft.github.io/presidio/
We need to make sure that all docs are properly referenced there and (if applicable) in the README.md

File: README.MD (hunk 1 above)
Comment (navalev) [reply README.MD:52]: After thinking about this I would have one deployment doc with all the options, and seperate them in the doc to "Production" and "Development". LMK if that seems ok and I will updae the doc

File: README.MD (hunk 2 above)
Comment (navalev) [reply README.MD:21]: is the overview file needed? I didn't see it's reference anywhere. Whated to reduce the amount of information in the main readme as it is hard to find information that way. I suggest updating the overview doc with most recent info and still referencing to it. WDYT?

File: README.MD (hunk 3 above)
Comment (navalev) [reply README.MD:25]: wanted to make the main readme minimal. I'll add this back, np

File: docs/KIND.md (hunk 4 above)
Comment (navalev) [reply docs/KIND.md:1]: in the main readme. But I'm now thinking on combing it with the deploy.md to have one unified md with prod/dev seperation within the file. LMK if this work

File: README.MD (hunk 1 above)
Comment (omri374) [reply README.MD:52]: Sounds good

File: README.MD (hunk 2 above)
Comment (omri374) [reply README.MD:21]: I think it can be deleted if it's not referenced anywhere

File: README.MD (hunk 3 above)
Comment (omri374) [reply README.MD:25]: So are we keeping the overview.md or not? I think at least one image is necessary on the readme. Many people prefer visual information over textual

File: docs/KIND.md (hunk 4 above)
Comment (omri374) [reply docs/KIND.md:1]: Sounds good.

File: README.MD (hunk 2 above)
Comment (navalev) [reply README.MD:21]: still debating this :) will update

File: README.MD (hunk 3 above)
Comment (navalev) [reply README.MD]: I kept the overview, added a link in the index and also added a new image in the main readme (I think too many images are a bit of overload...). Let me know how it looks now

File: README.MD (hunk 2 above)
Comment (navalev) [reply README.MD]: @omri374 did some more refactoring and kept the overview.md, it is referenced in the index and well as the main readme

File: README.MD (hunk 3 above)
Comment (omri374) [reply README.MD]: IMO the previous image was better. This one has too much details on it. WDYT?

File: docs/deploy.md [hunk 6]
```diff
@@ -207,3 +170,77 @@ for item in response:
...
+
+1. Navigate into `<root>\deployment` from command line.
+
+2. If You have helm installed, but havn't run `helm init`, execute [deploy-helm.sh](../deployment/deploy-helm.sh) in the command line. It will install `tiller` (helm server side) on your cluster, and grant it sufficient permissions.
+
+```sh
+deploy-helm.sh
+```
+
+3. Optional - Grant the Kubernetes cluster access to the container registry. This step can be skipped and the script will pull the container images from [Microsoft syndicates container catalog](https://azure.microsoft.com/en-in/blog/microsoft-syndicates-container-catalog/)
```
Comment (omri374) [inline docs/deploy.md:204]: Maybe note that this is optional only if you plan to use the default images (and not create ones yourself)

File: docs/index.md [hunk 7]
```diff
@@ -9,22 +9,28 @@ New to Presidio? Read this material to quickly get up and running.
...
 - [Database and storage scanner](tutorial_scheduler.md)
 - [Architecture](design.md)
 
 Development:
+
 - [Setting up a development environment](development.md)
 - [Adding custom fields](custom_fields.md)
 - [Best practices for developing new recognizers](developing_recognizers.md)
 - [Analyzer service tutorial](tutorial_analyzer.md)
 - [Calling the different services](tutorial_service.md)
```
Comment (omri374) [inline docs/index.md:26]: Could you please the new tutorial_connector.md to the list? https://github.com/microsoft/presidio/blob/master/docs/tutorial_connector.md

File: docs/overview.md [hunk 8]
```diff
@@ -41,19 +39,22 @@ Presidio uses OCR to detect text in images. It further allows the redaction of t
 Presidio accepts multiple sources and targets for data annonymization. Specifically:
 
 1. Storage solutions
-    * Azure Blob Storage
-    * S3
-    * Google Cloud Storage
+
+   - Azure Blob Storage
+   - S3
+   - Google Cloud Storage
```
Comment (omri374) [inline docs/overview.md:45]: Could you please add Azure Data Lake gen 2?

Comment (navalev) [reply docs/overview.md:46]: done

File: docs/index.md (hunk 7 above)
Comment (navalev) [reply docs/index.md:26]: done

File: docs/deploy.md (hunk 6 above)
Comment (navalev) [reply docs/deploy.md]: done

File: README.MD (hunk 3 above)
Comment (navalev) [reply README.MD]: updated with new gif image

**Pull request merged**